admin-api-server/src/analytics/controller/analytics.controller.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Post,
+  Query,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common';
+import {
+  MetricEntity,
+  MetricParams,
+  Resolution,
+} from '../../analytics/entity/analytics.entity';
+import { AnalyticsService } from '../service/analytics.service';
+import { enumFromStringValue } from 'src/utils';
+
+@Controller('analytics')
+export class AnalyticsController {
+  constructor(private analyticsService: AnalyticsService) {}
+
+  @Get('sales/priceVolume/snapshot')
+  async salesPriceVolume(
```

Comment:
I would just add an auth guard to all the endpoint

Reply:
Wow. Yes